dotCMS/src/integration-test/java/com/dotmarketing/factories/MultiTreeAPITest.java

```diff
@@ -1341,6 +1341,95 @@ public void shouldReplaceVariantMultiTree() throws Exception {
+        final Contentlet contentlet_2 = new ContentletDataGen(contentType.id()).nextPersisted();
+        final Contentlet contentlet_3 = new ContentletDataGen(contentType.id()).nextPersisted();
+
+        final Template template = new TemplateDataGen().body("body").nextPersisted();
+        final Folder folder = new FolderDataGen().nextPersisted();
+        final HTMLPageAsset page = new HTMLPageDataGen(folder, template).nextPersisted();
+        final Structure structure = new StructureDataGen().nextPersisted();
+        final Container container = new ContainerDataGen().maxContentlets(1).withStructure(structure, "").nextPersisted();
+
+        final String uniqueId = UUIDGenerator.shorty();
+
+        for (Variant variant : list(variantA, VariantAPI.DEFAULT_VARIANT)) {
+            new MultiTreeDataGen()
+                    .setPage(page)
+                    .setContainer(container)
+                    .setContentlet(contentlet_1)
+                    .setInstanceID(uniqueId)
+                    .setPersonalization(DOT_PERSONALIZATION_DEFAULT)
+                    .setTreeOrder(1)
+                    .setVariant(variant)
+                    .nextPersisted();
+
+            new MultiTreeDataGen()
+                    .setPage(page)
+                    .setContainer(container)
+                    .setContentlet(contentlet_2)
+                    .setInstanceID(uniqueId)
+                    .setPersonalization(DOT_PERSONALIZATION_DEFAULT)
+                    .setTreeOrder(1)
+                    .setVariant(variant)
+                    .nextPersisted();
+
+            new MultiTreeDataGen()
+                    .setPage(page)
+                    .setContainer(container)
+                    .setContentlet(contentlet_1)
+                    .setInstanceID(uniqueId)
+                    .setPersonalization(persona.getKeyTag())
+                    .setTreeOrder(1)
+                    .setVariant(variant)
+                    .nextPersisted();
+
+            new MultiTreeDataGen()
+                    .setPage(page)
+                    .setContainer(container)
+                    .setContentlet(contentlet_3)
+                    .setInstanceID(uniqueId)
+                    .setPersonalization(persona.getKeyTag())
+                    .setTreeOrder(1)
+                    .setVariant(variant)
+                    .nextPersisted();
+        }
+
+        APILocator.getMultiTreeAPI().copyVariantForPage(page.getIdentifier(), variantA.name(), variantB.name());
+
+        final List<MultiTree> multiTrees = getMultiTrees(page.getIdentifier());
+
+        assertEquals(12, multiTrees.size());
+
+        final List<MultiTree> multiTreesByVariant = APILocator.getMultiTreeAPI()
+                .getMultiTreesByVariant(page.getIdentifier(), variantB.name());
+
+        assertEquals(4, multiTreesByVariant.size());
+
+        /*final List<String> multiTreeContentlets = multiTrees.stream()
```

Comment:
remove this?

Reply:
done https://github.com/dotCMS/core/pull/23258/commits/be0dccb64465aa648a685a47be69411aa478f1b7#diff-f8c32e1b537585977097aa4816b8b0fb99b6b0df0ba259590ea652fbd66030abL1423